src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs

```diff
@@ -224,6 +232,94 @@ public int RemoveAdornments()
 
                 return VSConstants.S_OK;
             }
+
+            int IVsDocOutlineProvider.GetOutline(out IntPtr phwnd, out IOleCommandTarget? ppCmdTarget)
+            {
+                phwnd = default;
+                ppCmdTarget = null;
+
+                var textView = GetActiveTextView();
+                RoslynDebug.AssertNotNull(textView);
+                var textBuffer = textView.TextBuffer;
+                var isCSharpContentType = textBuffer.ContentType.IsOfType(ContentTypeNames.CSharpContentType);
+                var isVisualBasicContentType = textBuffer.ContentType.IsOfType(ContentTypeNames.VisualBasicContentType);
+
+                if (isCSharpContentType || isVisualBasicContentType)
```

Comment:
Love this. Can you add a comment explaining that we will get called for all roslyn content types (F#, Typescript, and Javascript) in addition to just C# and VB and only C# and VB has the LSP server set up.

Reply:
can we not have this check, and just attempt to get the doc symbols no matter what?  that way F#/TS can light as well? t hsi effectively filters them out, even if they wanted to participate.

Reply:
@CyrusNajmabadi as of right now, the document symbol request response is null for other file types, so we included this check so that the new document outline window wouldn't appear. I would be happy to remove this check but until this is updated the symbol tree will always be empty for these file types... is that the intended behavior?

Reply:
Check has been removed